prover/src/lib.rs

```diff
@@ -191,14 +199,28 @@ pub trait Prover {
     // PROVIDED METHODS
     // --------------------------------------------------------------------------------------------
 
+    /// Builds the auxiliary trace.
+    #[allow(unused_variables)]
+    fn build_aux_trace<E, Hasher>(
+        &self,
+        main_trace: &ColMatrix<E::BaseField>,
+        transcript: &mut impl RandomCoin<BaseField = E::BaseField, Hasher = Hasher>,
+    ) -> AuxTraceWithMetadata<E, AuxRandElementsProver<Self, E>, AuxProofProver<Self>>
+    where
+        E: FieldElement,
+        Hasher: ElementHasher<BaseField = E::BaseField>,
+    {
+        unimplemented!("`Prover::build_aux_trace` needs to be implemented when the trace has an auxiliary segment.")
+    }
+
     /// Returns a STARK proof attesting to a correct execution of a computation defined by the
-    /// provided trace.
+    /// provided trace. The trace must not contain an auxiliary trace segment.
```

Comment:
I think the comment about auxiliary trace is no longer relevant. Also, maybe worth mentioning that `Proof` can now contain additional stuff.

Reply:
Done. Docstring now briefly mentions the aux proof, refers to `Proof` for more information